### PR TITLE
Fix!(spark): parse computed column constraints appropriately

### LINKED ISSUE #2301: bug: Databricks COMPUTED column is parsed as a `GENERATEDASIDENTITYCOLUMNCONSTRAINT`
Databricks COMPUTED column is parsed as a `GENERATEDASIDENTITYCOLUMNCONSTRAINT` causing computed column to be rendered as IDENTITY column

**Fully reproducible code snippet**
```python
>>> import sqlglot

>>> sqlglot.parse_one('CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(y)))',read='databricks')
(CREATE this: 
  (SCHEMA this: 
    (TABLE this: 
      (IDENTIFIER this: foo, quoted: False)), expressions: 
    (COLUMNDEF this: 
      (IDENTIFIER this: x, quoted: False), kind: 
      (DATATYPE this: Type.INT, nested: False, prefix: False), constraints: 
      (COLUMNCONSTRAINT kind: 
        (GENERATEDASIDENTITYCOLUMNCONSTRAINT this: True, expression: 
          (YEAR this: 
            (TSORDSTODATE this: 
              (COLUMN this: 
                (IDENTIFIER this: y, quoted: False)))))))), kind: TABLE, exists: False)


>>> sqlglot.transpile('CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(y)))', read='databricks',write='tsql')
['CREATE TABLE foo (x INTEGER IDENTITY(1, 1))']
```

**Official Documentation**
https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html

```
{ { [CREATE OR] REPLACE TABLE | CREATE [EXTERNAL] TABLE [ IF NOT EXISTS ] }
  table_name
  [ table_specification ]
  [ USING data_source ]
  [ table_clauses ]
  [ AS query ] }

table_specification
  ( { column_identifier column_type [ NOT NULL ]
      [ GENERATED ALWAYS AS ( expr ) |
        GENERATED { ALWAYS | BY DEFAULT } AS IDENTITY [ ( [ START WITH start ] [ INCREMENT BY step ] ) ] |
        DEFAULT default_expression ]
      [ COMMENT column_comment ]
      [ column_constraint ] } [, ...]
    [ , table_constraint ] [...] )

table_clauses
  { OPTIONS clause |
    PARTITIONED BY clause |
    clustered_by_clause |
    LOCATION path [ WITH ( CREDENTIAL credential_name ) ] |
    COMMENT table_comment |
    TBLPROPERTIES clause } [...]

clustered_by_clause
  { CLUSTERED BY ( cluster_column [, ...] )
    [ SORTED BY ( { sort_column [ ASC | DESC ] } [, ...] ) ]
    INTO num_buckets BUCKETS }
```

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #2301

References:
- https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html
- https://learn.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16
- https://learn.microsoft.com/en-us/sql/relational-databases/tables/specify-computed-columns-in-a-table?view=sql-server-ver16
- https://learn.microsoft.com/en-us/sql/t-sql/functions/cast-and-convert-transact-sql?view=sql-server-ver16
- https://github.com/apache/spark/pull/38823

### CODE REVIEW

File: tests/dialects/test_databricks.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def test_databricks(self):
             "CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(y)))",
             write={
                 "databricks": "CREATE TABLE foo (x INT GENERATED ALWAYS AS (YEAR(TO_DATE(y))))",
+                "tsql": "CREATE TABLE foo (x INTEGER AS YEAR(CAST(y AS DATE)))",
```
Comment (GeorgeSittas) [inline tests/dialects/test_databricks.py:35]: Double-checking if this is correct in T-SQL..

**Review by tobymao — APPROVED** [review]

File: tests/dialects/test_databricks.py (hunk 1 above)
Comment (GeorgeSittas) [reply tests/dialects/test_databricks.py:35]: It looks like we need to drop `INTEGER` here, otherwise it's a syntax error, cc @dmoore247

Comment (GeorgeSittas) [reply tests/dialects/test_databricks.py:35]: Just verified that the following statement is valid T-SQL.

```sql
CREATE TABLE foo (x DATETIME, y AS YEAR(CAST(x AS DATE)))
```

File: sqlglot/generator.py [hunk 2]
```diff
@@ -207,6 +207,9 @@ class Generator:
     # The keyword used to clone a table in a DDL statement
     CLONE_KEYWORD = "CLONE"
 
+    # Whether or not to incldue the type of a computed column in the CREATE DDL
```
Comment (GeorgeSittas) [inline sqlglot/generator.py:210]:
```suggestion
    # Whether or not to include the type of a computed column in the CREATE DDL
```

**Pull request merged**